packages/docusaurus-plugin-content-docs/src/globalData.ts

```diff
@@ -20,6 +25,29 @@ export function toGlobalDataDoc(doc: DocMetadata): GlobalDoc {
   };
 }
 
+export function toGlobalSidebars(
+  sidebars: Sidebars,
+  version: LoadedVersion,
+): Record<string, GlobalSidebar> {
+  const {getFirstDocOfSidebar} = createSidebarsUtils(sidebars);
+  return mapValues(sidebars, (sidebar, sidebarId) => {
+    const firstDoc = getFirstDocOfSidebar(sidebarId);
+    if (!firstDoc) {
+      throw new Error(
+        `The sidebar "${sidebarId}" doesn't seem to contain any documentation`,
+      );
+    }
+    return {
+      link: firstDoc.isAutoGeneratedIndex
+        ? normalizeUrl([version.versionPath, firstDoc.slug])
+        : version.docs.find(
+            (doc) =>
+              doc.id === firstDoc.id || doc.unversionedId === firstDoc.id,
+          )!.permalink,
```

Comment:
`link` should rather be an object containing both a `path` + `label`. Using an object is more extensible for possible future use-cases

Reply:
I don't quite understand. How would the docs plugin control the label of the link? Are we going to have something like `label` in `sidebars.js` in the future? In all cases, the label is either the navbar item label or sidebar ID, both of which the theme already know of from the navbar item definition and don't need to be passed down from the global data. I think `link` as a string is also an intuitive API.

Reply:
The default/fallback label can be the label of the linked item. Either the doc label, or the autogenerated sidebar category label.

The sidebarId is not really meant to be a human label, so it's not a very good fallback. But I do agree that my proposal is not a perfect fallback either, it's just a little bit better.

Imagine this screen had a category link on Episodes.

![image](https://user-images.githubusercontent.com/749374/147647551-53406fd7-5258-4d9b-8726-531fd58c4978.png)

Then linking to this sidebar would automatically produce an "Episodes" navbar label

If there's no category link, then the navbar label could automatically become "Episode 10". 

I find this interesting to allow dynamic patterns where the navbar label auto-adjust to sidebar content, but I agree this is a quite niche use case and many people would prefer to specify their own label.

I'm open to discussing and implementing this as part of a separate PR if you feel it's not something we really need

But I'd rather replace `link: string` by `link: {path: string}`

Reply:
Oh, so we want to display the sidebar item's label as fallback? I'm okay with that, and if that's what you meant, then `{path, label}` also makes sense

Reply:
Yes, as we are always linking to some sidebar item, there's always a fallback label that we can use: the item label.

That may not work well for all use-cases but might be useful for some edge-cases. The sidebarId fallback is more likely to never be a very good solution.